src/applications/edu-benefits/components/SchoolSelectField.jsx

```diff
@@ -36,6 +36,15 @@ export class SchoolSelectField extends React.Component {
     this.debouncedSearchInstitutions = _.debounce(
       value => this.props.searchSchools(value),
       150);
+
+  }
+
+  componentDidMount() {
+    // hydrate search if restoring from SiP
+    const searchTermToRestore = this.props.formData['view:searchString'];
+    if (searchTermToRestore && searchTermToRestore !== this.props.institutionQuery) {
```

Comment:
Kick off an action if there's a search term on the `formData` that doesn't match what's in the current state.